Automattic/wp-calypso
# Fix the clean:public script to do a better job cleaning public/ folder

Fixes a bug with the following steps to reproduce:
1. Do a production build with `NODE_ENV=production CALYPSO_ENV=production npm run build`
2. Clean the build artifacts with `npm run clean`

**Actual result:** some files that should have been clean aren't:
- many `public/images/*.{jpg,png}` files created by the webpack file loader
- `*.css.map` files in `public/sections` and `public/sections-rtl`
- the `public/sections[-rtl]` directories should actually exist at all after cleaning
- the `public/vendors~map` directory with a [3rd party library used](https://github.com/Automattic/wp-calypso/blob/225189a65dd4825b2d37d8081e56242326dc7663/client/gutenberg/extensions/map/component.js#L263-L264) by Map Gutenblock

Some of the files are not even Gitignored. This PR makes the clean complete. The task is a bit tricky because some files in the `public/` directory are version controlled, some aren't. Namely, files directly in `public/` and `public/images/` are build artifacts, but files in their subdirectories are version controlled. We need to be careful when deleting files from there.

**Patch SEO Preview Upgrade Nudge**
File `public/advanced-seo-nudge.png` breaks the rules -- it's a version controlled file that's directly in `public/`. That's why f9bc843 moves it to a new place and makes it a webpack asset. The commit also updates the server-side webpack config to ignore PNG and other files.

**Disable vendors prefix for the map/mapbox-gl chunk**
The files in `public/vendors~map` belong to the Map Gutenblock and the directory name should really be `map` rather than `vendors~map`. c5594c9 changes that by doing some webpack config magic. @tyxla @sirreal please test that the blocks built for Jetpack correctly place the Map assets into `map/` directory.

**Do the boring updates in the clean script and `.gitignore`**
The rest is easy.

#### Testing instructions

Go to a preview of a post on a non-Business site and select "Search&Social" in the dropdown menu. Verify that instead of your post there are cookies:
<img width="568" alt="screenshot 2018-12-12 at 15 08 18" src="https://user-images.githubusercontent.com/664258/49874866-ea1cd700-fe1f-11e8-9c12-b1972f6cd828.png">

And of course, run `npm run clean` and verify things get cleaned up properly.

`gutenpack-jn`
